### PR TITLE
Croissants

### PR DESCRIPTION
**Pull request by Shadowmech88**
Code and sprite scavenged from #10539.

:cl:
 * rscadd: Added croissants, made with 5u water, 5u flour, 5u milk, and one egg in the microwave.

### CODE REVIEW

**Comment by kol1th** [comment]
ayy lmao

[UPDATE WIKI] tag when @Probe1

**Comment by Probe1** [comment]
Update wiki then merge.

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 1]
```diff
@@ -4832,3 +4832,13 @@
...
 		Move(get_turf(O))
+
+/obj/item/weapon/reagent_containers/food/snacks/croissant
+	name = "croissant"
+	desc = "True French cuisine."
+	icon_state = "croissant"
+
+/obj/item/weapon/reagent_containers/food/snacks/croissant/New()
+	..()
+	reagents.add_reagent("nutriment", 10)
```
Comment (Kurfursten) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: This is a lot, maybe scale it back to 6 like the baguette.

**Comment by Rattlederp** [comment]
Wiki is updated

**Comment by Probe1** [comment]
Do what Kurf said so I can merge.

**Comment by Shadowmech88** [comment]
Done.

**Comment by ComicIronic** [comment]
@Probe1 Shouldn't have merged this, it doesn't use the reagent defines.

@Shadowmech88 Open a new PR with the define changes.

**Comment by Probe1** [comment]
What are you talking about?

**Comment by ComicIronic** [comment]
We have defines for reagent names so the compiler catches typos, and the code in this PR doesn't use them. e.g. `FLOUR` instead of `"flour"`.

**Comment by Kurfursten** [comment]
The compiler doesn't actually catch defines typos, that's just wishful thinking.

untitled

**Comment by Exxion** [comment]
Alright
Now THAT'S weird

**Comment by Exxion** [comment]
Oh.
That's right.
BYOND doesn't do any kind of checking at all of args to procs at compile time
Because why WOULD it?

**Comment by PJB3005** [comment]
@Kurfursten I have no idea what's going on in that pic, but it definitely does.

**Comment by PJB3005** [comment]
Oh I see.

It's because BYOND is fucking retarded and sees the `FLUOR` in this case as doing a positional argument.

Jesus christ BYOND.

**Comment by ComicIronic** [comment]
It would still catch a typo in `"nutriment"` in this case. We have the defines for a reason, and we should be using them.

**Comment by Exxion** [comment]
I'm pretty sure it wouldn't. Still an arg.
But we should still be using it for the sake of consistency

**Comment by ComicIronic** [comment]
No, the list case only happens because Byond thinks you're trying to name an argument (because you can use any valid identifier as a list index without making it a string because ???) - passing an argument *value* is still checked, because the value has to be an evaluatable expression, and a typo'd identifier can't be evaluated to anything.

**Comment by Exxion** [comment]
Alright, fair enough

**Pull request merged**